xpallets/assets/rpc/src/lib.rs

```diff
@@ -96,66 +95,33 @@ where
         api.assets(&at)
             .map(|map| {
                 map.into_iter()
-                    .map(|(id, info)| (id, info.into()))
+                    .map(|(id, info)| {
+                        // if balance not use u128, this part could be deleted
+                        let mut r = BTreeMap::new();
+                        AssetType::iter().for_each(|type_| {
+                            let balance = if let Some(b) = info.balance.get(type_) {
+                                *b
+                            } else {
+                                Balance::zero()
+                            };
+                            r.insert(*type_, balance);
+                        });
```

Comment:
```suggestion
                        let balance = AssetType::iter()
                             .map(|ty| {
                                 (
                                     *ty,
                                     info.balance.get(ty).map(|b| *b).unwrap_or(Balance::zero()),
                                 )
                             })
                             .collect::<BTreeMap<_, _>>();
```

Personally I would prefer implementing `TotalAssetInfo` from `info` since only the `balance` field needs to be tranformed.